src/taipy/gui_core/_context.py

```diff
@@ -271,11 +298,110 @@ def submit_entity(self, state: State, id: str, action: str, payload: t.Dict[str,
+                                _SubmissionStatus.SUBMITTED,
+                                job_ids,
+                            )
+                        self.scenario_status_callback(jobs[0].id if isinstance(jobs, list) else jobs.id)
+                    else:
+                        _warn(f"on_submission_change(): '{action}' is not a valid function.")
                 state.assign(_GuiCoreContext._SCENARIO_VIZ_ERROR_VAR, "")
             except Exception as e:
                 state.assign(_GuiCoreContext._SCENARIO_VIZ_ERROR_VAR, f"Error submitting entity. {e}")
 
+    def _get_submittable_status(self, jobs_ids: t.List[str]):
+        status = _SubmissionStatus.UNDEFINED
+        blocked = False
+        waiting = False
+        running = False
+        completed = False
+        for id in jobs_ids:
+            job = core_get(id)
+            if not job:
+                continue
+            if job.is_failed():
+                status = _SubmissionStatus.FAILED
+                break
+            if job.is_canceled():
+                status = _SubmissionStatus.CANCELED
+                break
+            if not blocked and job.is_blocked():
+                blocked = True
+            if not waiting and job.is_pending():
+                waiting = True
+            if not running and job.is_running():
+                running = True
+            if not completed and (job.is_completed() or job.is_skipped()):
+                completed = True
+        if status is _SubmissionStatus.UNDEFINED:
+            if waiting:
+                status = _SubmissionStatus.WAITING
+            elif blocked:
+                status = _SubmissionStatus.BLOCKED
+            elif running:
+                status = _SubmissionStatus.RUNNING
+            elif completed:
+                status = _SubmissionStatus.COMPLETED
+        return status
+
+    def scenario_status_callback(self, job_id: str):
+        if not job_id:
+            return
+        try:
+            job = core_get(job_id)
+            if not job:
+                return
+            sub_id = job.submit_id
+            res = self.client_jobs_by_submission.get(sub_id)
+            if not res:
+                return
+
+            client_id, module_context, action, entity_id, current_status, jobs = res
+            if not action or not client_id or not entity_id or not jobs:
+                return
+
+            entity = core_get(entity_id)
+            if not entity:
+                return
+
+            on_action_function = self.gui._get_user_function(action)
+            if not callable(on_action_function):
+                return
+
+            new_status = self._get_submittable_status(jobs)
+            if current_status is not new_status:
+                # callback
+                self.gui._call_user_callback(
+                    client_id,
+                    on_action_function,
+                    [entity, {"submission_status": new_status.name, "job": job}],
```

Comment:
I think the first parameter should be... something else.
In the future, that will be a 'Submission'. Maybe we need to create a temporary wrapper class for that.

Reply:
something like ?

Reply:
expose a submission class from taipy ?
I would think it should come from core ?